recipes/libcublas/meta.yaml

```diff
@@ -0,0 +1,108 @@
+{% set name = "libcublas" %}
+{% set version = "12.0.1.189" %}
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 38272cc9184b4a57a6250fc2cd914ffc827f9f8cd5a3e1356bc3c31f843caf8f  # [linux64]
+  sha256: 2d46e735f8e76e2eb7df1b4f41d56aa8d5d2683f99988bac82d6ae33fb6d344b  # [ppc64le]
+  sha256: 96b68f6821684f39f338862639ce74a74c75830bf44dcd17ef372a1cb8c2304d  # [aarch64]
+  sha256: 88f29bec81880844da2eee13f1cdc51e1ccd8b4c60af6df633577810b30f80d0  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+
+outputs:
+  - name: libcublas
+    requirements:
+      host:
+         - cuda-version {{ cuda_version }}
+      run:
+         - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+    files:
+      - lib/lib*blas*.so.*            # [linux]
+      - targets/*/lib/lib*blas*.so.*  # [linux]
+      - Library\bin\cublas*.dll                   # [win]
+      - Library\bin\nvblas*.dll                   # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcublas.so.*  # [unix]
+        - if not exist %LIBRARY_LIB%\\cublas.lib exit 1  # [win]
+
+  - name: libcublas-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcublas", max_pin="x") }}
+    requirements:
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - libcublas >={{ version }}
+      run_constrained:
+        - libcublas-static >={{ version }}
+    files:
+      - lib/lib*blas*.so            # [linux]
+      - targets/*/lib/lib*blas*.so  # [linux]
+      - lib/stubs                   # [linux]
+      - include                     # [linux]
+      - targets/*/include/*         # [linux]
+      - lib/pkgconfig               # [linux]
+      - src
```

Comment:
They are being copied by the build script, and dropping them isn't an option. What would a more correct location be?

`2023-03-22T09:14:30.6165931Z ‘src/fortran.c’ -> ‘$PREFIX/src/fortran.c’`

Reply:
I don't think we saw the `src` files being shipped in the artifacts being produced? (I need to look back and confirm that.) I don't know what a more correct location would be. @adibbley Can you explain how these are used, and perhaps what search paths are considered? I'm not familiar with using `$PREFIX/src` in other conda packages and am not sure what precedents to compare against.

Reply:
Update: I do see the `src/` files in the artifacts being produced. I guess we can just ship those files in `$PREFIX/src`, unless anyone has other ideas.

Reply:
@adibbley The concern during the discussion was that the shipped filenames (ex: `fortran.c`) are hard to tell if they're coming from `libcublas` or another package, and could potentially be clobbering. Also, it's almost impossible for conda-forge users to locate these files, unless the cuBLAS team explicitly documents about `$PREFIX/src` (or wherever they'd be). So it's of very little use. I don't have strong opinions on this, though.

Reply:
The cublas docs do mention them in `src` specifically: https://docs.nvidia.com/cuda/cublas/#cublas-fortran-bindings
Maybe a `$PREFIX/cuda/src/` would make sense here? We probably have some other files that might be worth putting in a `cuda` directory for clarity. I'm thinking things like `docs` `samples` and the like could make more sense under `$PREFIX/cuda`.

Reply:
Okay. My primary complaint was that the files are very generically named -- `src/fortran.c` could be literally anything, it has no indication of a relationship to CUDA or cuBLAS.

I kind of like the idea of moving these files into `$PREFIX/cuda/{doc,samples,src}` so that CUDA-related files (besides include/lib) are together.

For comparison: In my primary conda environment (which has a bunch of assorted packages), I see a `doc` directory with a couple things from Qt and CMake, but nothing else. It would feel odd to put extensive documentation from CUDA in that directory, I think. There are also directories like `share` but that doesn't feel quite right either. Since I don't see a better candidate, I'm a weak +1 for `$PREFIX/cuda` for all these files.